src/vmm/src/devices/virtio/iovec.rs

```diff
@@ -35,19 +35,27 @@ type IoVecVec = SmallVec<[iovec; 4]>;
 /// It describes a buffer passed to us by the guest that is scattered across multiple
 /// memory regions. Additionally, this wrapper provides methods that allow reading arbitrary ranges
 /// of data from that buffer.
-#[derive(Debug)]
+#[derive(Debug, Default)]
 pub struct IoVecBuffer {
     // container of the memory regions included in this IO vector
     vecs: IoVecVec,
```

Comment:
Mh, now that we're reusing iovec buffers, I was hoping that the `IoVecVec`/`smallvec` stuff won't be needed anymore. Can we revert https://github.com/firecracker-microvm/firecracker/commit/e5a5d0e02b3d5941ada4f48891b836507cce1dea on top of this and run some performance tests to validate that hope? It'd simplify the code nicely

Reply:
Oh sure I can give that a try that would be nice.

Reply:
As per offline discussion: I mixed up some things here, to avoid the smallvec stuff we'd need to use this in the vsock device, not the net device.